dm/syncer/util.go

```diff
@@ -153,6 +156,43 @@ func subtaskCfg2BinlogSyncerCfg(cfg *config.SubTaskConfig, timezone *time.Locati
 		}
 	}
 
+	var rowsEventDecodeFunc func(*replication.RowsEvent, []byte) error
+	if baList != nil {
+		// we don't track delete table events, so simply reset the cache if it's full
+		allowListCache := make(map[uint64]struct{}, maxCapacity)
+		blockListCache := make(map[uint64]struct{}, maxCapacity)
+
+		rowsEventDecodeFunc = func(re *replication.RowsEvent, data []byte) error {
+			pos, err := re.DecodeHeader(data)
+			if err != nil {
+				return err
+			}
+			if _, ok := blockListCache[re.TableID]; ok {
+				return nil
+			} else if _, ok := allowListCache[re.TableID]; ok {
+				return re.DecodeData(pos, data)
+			}
+
+			tb := &filter.Table{
+				Schema: string(re.Table.Schema),
+				Name:   string(re.Table.Table),
+			}
+			if skipByTable(baList, tb) {
+				if len(blockListCache) >= maxCapacity {
+					blockListCache = make(map[uint64]struct{}, maxCapacity)
```

Comment:
I will prefer to evict one record rather than reset the whole cache. Maybe an LRU or CLOCK cache is more appropriate.

But it might be rare to run out of capacity, so it's okay.

Reply:
add a TODO [c5511e7](https://github.com/pingcap/tiflow/pull/7622/commits/c5511e714c3b1724f4c106c3d0b63f8a027866b1)